src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -6583,6 +6606,231 @@ HRESULT ProfToEEInterfaceImpl::GetDynamicFunctionInfo(FunctionID functionId,
+ *      reJitId              - The ReJIT id.
+ *      cCodeStartAddresses  - A parameter for indicating the size of buffer for the codeStartAddresses parameter.
+ *      pcCodeStartAddresses - An optional parameter for returning the true size of the codeStartAddresses parameter.
+ *      codeStartAddresses   - The array to be filled up with native code addresses.
+ * 
+ * Returns:
+ *   S_OK if successful
+ *
+ */
+HRESULT ProfToEEInterfaceImpl::GetNativeCodeStartAddresses(FunctionID functionID, 
+                                                           ReJITID reJitId, 
+                                                           ULONG32 cCodeStartAddresses, 
+                                                           ULONG32 *pcCodeStartAddresses, 
+                                                           UINT_PTR codeStartAddresses[])
+{
+    CONTRACTL
+    {
+        NOTHROW;
+        GC_NOTRIGGER;
+        MODE_ANY;
+        EE_THREAD_NOT_REQUIRED;
+        CAN_TAKE_LOCK;
+
+        SO_NOT_MAINLINE;
+
+        PRECONDITION(CheckPointer(pcCodeStartAddresses, NULL_OK));
+        PRECONDITION(CheckPointer(codeStartAddresses, NULL_OK));
+    }
+    CONTRACTL_END;
+
+    if (functionID == NULL)
+    {
+        return E_INVALIDARG;
+    }
+
+    PROFILER_TO_CLR_ENTRYPOINT_ASYNC_EX(kP2EEAllowableAfterAttach,
+    (LF_CORPROF,
+        LL_INFO1000,
+        "**PROF: GetNativeCodeStartAddresses 0x%p 0x%p.\n",
+        functionID, reJitId));
+    
+    HRESULT hr = S_OK;
+   
+    EX_TRY
+    {
+        if (pcCodeStartAddresses != NULL)
+        {
+            *pcCodeStartAddresses = 0;
+        }
+
+        MethodDesc * methodDesc = FunctionIdToMethodDesc(functionID);
+        PTR_MethodDesc pMD = PTR_MethodDesc(methodDesc);
+        ULONG32 trueLen = 0;
+        StackSArray<UINT_PTR> addresses;
+
+        CodeVersionManager *pCodeVersionManager = pMD->GetCodeVersionManager();
+
+        ILCodeVersion ilCodeVersion = NULL;
+        {
+            CodeVersionManager::TableLockHolder lockHolder(pCodeVersionManager);
+
+            ilCodeVersion = pCodeVersionManager->GetILCodeVersion(pMD, reJitId);
+            
+            NativeCodeVersionCollection nativeCodeVersions = ilCodeVersion.GetNativeCodeVersions(pMD);
+            for (NativeCodeVersionIterator iter = nativeCodeVersions.Begin(); iter != nativeCodeVersions.End(); iter++)
+            {
+                addresses.Append((*iter).GetNativeCode());
+
+                ++trueLen;
+            }
+        }
+
+        if (pcCodeStartAddresses != NULL)
+        {
+            *pcCodeStartAddresses = trueLen;
+        }
+
+        if (codeStartAddresses != NULL)
+        {
+            if (cCodeStartAddresses < trueLen)
+            {
+                hr = HRESULT_FROM_WIN32(ERROR_INSUFFICIENT_BUFFER);
+            }
+            else
+            {
+                for(ULONG32 i = 0; i < cCodeStartAddresses; ++i)
+                {
+                    codeStartAddresses[i] = addresses[i];
```

Comment:
You've got a buffer overrun here. cCodeAddresses can be larger than trueLen but addresses only has length trueLen.